### PR TITLE
Remove Symbol polyfill in module namespaces

### LINKED ISSUE #3097: Simplify Symbol.toStringTag for module namespaces
There is currently no way for Rollup to generate simple output of the `toStringTag` in module namespaces. It always uses a verbose polyfill. Ideally we should always generate simple output, or make the polyfill optional.

As configuration options are not available in the REPL, I'm posting the sample code here.

```javascript
// main.mjs 

import * as nsA from "./main.mjs";
console.log(nsA);
```

### Expected Behavior / Situation

Produce namespaces that has `Symbol.toStringTag` set and not have the polyfill behavior.

```javascript
// expected bundle.mjs

var nsA = /*#__PURE__*/Object.freeze({
    [Symbol.toStringTag]: 'Module'
});

console.log(nsA);
```

### Actual Behavior / Situation

With `namespaceToStringTag` option enabled, Rollup produces a polyfill for the environments that don't have `Symbol.toStringTag` available. 

```sh
npx rollup \
    --input main.mjs \
    --format esm \
    --namespaceToStringTag \
    --output.file bundle.mjs
```

```javascript
// bundle.mjs

var nsA = {
};
if (typeof Symbol !== 'undefined' && Symbol.toStringTag)
    Object.defineProperty(nsA, Symbol.toStringTag, { value: 'Module' });
else
    Object.defineProperty(nsA, 'toString', { value: function () { return '[object Module]' } });
/*#__PURE__*/Object.freeze(nsA);

console.log(nsA);
```

### Modification Proposal

We can either:
- make `output.namespaceToStringTag` option not produce the polyfill
- introduce a new configuration option that assumes `Symbol.toStringTag` is available

I would be happy to work on either of these proposals or any other approach you propose.

### PR DESCRIPTION
**Pull request by mkubilayk**
This PR contains:
- [ ] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [x] other

Are tests included?
- [x] yes (*bugfixes and features will not be merged without tests*)
- [ ] no

Breaking Changes?
- [x] yes (*breaking changes will not be merged unless absolutely necessary*)
- [ ] no

List any relevant issue numbers:

Fixes #3097.

### Description

This simplifies generated code for module namespaces by removing the `Symbol` polyfill. It is technically a breaking change but it was agreed in https://github.com/rollup/rollup/issues/3097#issuecomment-533856451.

### CODE REVIEW

File: src/ast/variables/NamespaceVariable.ts [hunk 1]
```diff
@@ -91,25 +91,17 @@ export default class NamespaceVariable extends Variable {
...
-		if (options.namespaceToStringTag) {
-			output += `${n}if${_}(typeof Symbol${_}!==${_}'undefined'${_}&&${_}Symbol.toStringTag)${n}`;
-			output += `${t}Object.defineProperty(${name},${_}Symbol.toStringTag,${_}{${_}value:${_}'Module'${_}});${n}`;
-			output += `else${n || ' '}`;
-			output += `${t}Object.defineProperty(${name},${_}'toString',${_}{${_}value:${_}function${_}()${_}{${_}return${_}'[object Module]'${
-				options.compact ? ';' : ''
-			}${_}}${_}});${n}`;
-			output += `${callee}(${name});`;
-		}
+		let output = `${options.varOrConst} ${name} = ${`${callee}({${n}${members.join(
```
Comment (lukastaegert) [inline src/ast/variables/NamespaceVariable.ts]: For the compact output, there are still a few empty spaces that can be replaced with `${_}`, which will be replaced with an empty string for `compact: true` output (around the `=`). Also, is the template string inside the template string intended?

Comment (lukastaegert) [reply src/ast/variables/NamespaceVariable.ts]: But I see the `${_}` were already missing in the original form.

Comment (mkubilayk) [reply src/ast/variables/NamespaceVariable.ts]: Yeah, nested template string is intended and were there already. You are right,  `${_}` were missing but should I update?

Comment (lukastaegert) [reply src/ast/variables/NamespaceVariable.ts]: Please do!

Comment (lukastaegert) [reply src/ast/variables/NamespaceVariable.ts]: Also if getting rid of the nested template string is possible, that would be nice. Looks like this to me.

**Comment by mkubilayk** [comment]
@lukastaegert Thanks for the review. Incorporated your feedback.

**Comment by Havunen** [comment]
Nice feature. Documentation missing tho! :)

**Pull request merged**